src/cgnslib.c

```diff
@@ -1325,63 +1324,147 @@ int cg_family_read(int file_number, int B, int F, char *family_name,
+    /* Init tokenization loop:
+     * We work with pointer to variables to simplify switch from cgns_base structure
+     * variables to cgns_family structure variables.
+     *  */
+    int* nfamilies_p = &(base->nfamilies);
+    cgns_family** family_p = &(base->family);
+    double parent_id = base->id;
+
+    /* Start loop over token */
+    pch = strtok( family_name_path, "/" );
+    while( pch ) {
+
+        tok = pch;
+        pch = strtok (NULL, "/"); /* when pch is null, we are at leaf (last token)*/
+
+        /* Check token size, should be max 32 */
+        if (cgi_check_strlen(tok)) {
+            cgi_error( "Invalid Family_t node %s", tok );
+            return CG_ERROR;
+        }
+        skip = 0; /* Flag to skip node creation */
+        /* Look for token in existing Family_t childs */
+        for( index=0; index < (*nfamilies_p); index++ ) {
+
+            /* If found existing Family_t named as token within the Family_t childs */
+            if( strcmp( tok, (*family_p)[index].name) == 0 ) {
+
+                /* If last token, then overwriting action */
+                if( !pch ) {
+
+                    /* Overwriting action not allowed on pure write mode */
+                    if( cg->mode == CG_MODE_WRITE ) {
+                        cgi_error( "Duplicate child name found: %s", tok );
+                        return CG_ERROR;
+                    }
+                    /* Modify mode : overwrite an existing family */
+                    if( cgi_delete_node( parent_id, (*family_p)[index].id )) {
+                        return CG_ERROR;
+                    }
+                    /* Save the old in-memory address to overwrite */
+                    family = &( (*family_p)[index] );
+                    /* free memory */
+                    cgi_free_family(family);
+                    break; /* quit "for" loop */
+
+                }
+                /* else, progressing in family tree path */
+                else {
+                    family = &( (*family_p)[index] ); /* ?? */
+                    skip = 1; /* intermediate node exists and should not be created or overwritten */
+                    break;
+                }
+
+            }
+        }
+
+        /* ... or add a Family_t Node */
+        if( index == *nfamilies_p ) {
+            if( *nfamilies_p == 0 ) {
+                *family_p = CGNS_NEW( cgns_family, (*nfamilies_p)+1 );
+            } else {
+                *family_p = CGNS_RENEW( cgns_family, (*nfamilies_p)+1, *family_p );
+            }
+            family = &( (*family_p)[*nfamilies_p] );
+            (*nfamilies_p)++;
+        }
+
+        (*F) = index+1;
+
+        if( ! skip ) { /* If not an existing intermediate family node */
+            memset( family, 0, sizeof(cgns_family) );
+            strcpy( family->name, tok );
+
+            /* Save data in file */
+            if( cgi_new_node( parent_id, tok, "Family_t", &family->id, "MT", 0, 0, 0) ) {
+                return CG_ERROR;
```

Comment:
Probably we should change all theses "*_t" magic values by pre-processing constants one of these days...

Reply:
Let's do it in a separate branch to not infect this topic